ocs_ci/utility/aws.py

```diff
@@ -1134,3 +1134,38 @@ def update_config_from_s3(bucket_name=constants.OCSCI_DATA_BUCKET, filename=cons
     # set in config and store it for that scope
     config.update(config_yaml)
     return config_yaml
+
+
+def delete_cluster_buckets(cluster_name):
+    """
+    Delete s3 buckets corresponding to a particular OCS cluster
+
+    Args:
+        cluster_name (str): name of the cluster the buckets belong to
+
+    """
+    region = config.ENV_DATA['region']
+    base_domain = config.ENV_DATA['base_domain']
+    s3_client = boto3.client('s3', region_name=region)
+    buckets = s3_client.list_buckets()['Buckets']
+    bucket_names = [bucket['Name'] for bucket in buckets]
+    logger.debug("Found buckets: %s", bucket_names)
+
+    # patterns for mcg target bucket and image-registry buckets
+    patterns = [
+        f"nb.(\\d+).apps.{cluster_name}.{base_domain}",
+        f"{cluster_name}-(\\w+)-image-registry-{region}-(\\w+)"
+    ]
+    for pattern in patterns:
+        r = re.compile(pattern)
+        filtered_buckets = list(filter(r.search, bucket_names))
+        if len(filtered_buckets) == 1:
+            s3_resource = boto3.resource('s3', region_name=region)
+            bucket_name = filtered_buckets[0]
+            logger.warning("Deleting all files in bucket %s", bucket_name)
```

Comment:
shouldn't it be logger.info?

Reply:
can be fixed later, merging

Reply:
I opted for warning here because the deletion is destructive. If we don't really care since this is during cleanup anyways I can change it to info level.